src/pages/ProgressReport.vue

```diff
@@ -529,6 +533,14 @@ const exportAll = async () => {
   );
 };
 
+const optionalAssessments = computed(() => {
+  return administrationInfo.value?.assessments.filter((assessment) => assessment.optional) || [];
+});
+
+const isOptional = (_taskId) => {
+  return optionalAssessments.value.some((assessment) => assessment.taskId === _taskId);
+};
+
```

Comment:
I think we can get rid of this part since we want to evaluate the optional status on a per-assignment (not per administration) basis. That is, assessments can be optional for some and required for others.

```suggestion
```